feature-reference.md

```diff
@@ -1,14 +1,155 @@
+
+Normally, you will have a large chunk (80%+) of traces in the category "other" (which means they
+are quite rare individually, and are grouped into a big chunk), and a few hundred stack traces
+shown in the "Top XXX" list below.
+
+The UI subdivides the time interval that you are looking at into smaller chunks, and for each time
+chunk Prodfiler retrieves the top-50 most common stack traces. These are then displayed in the
+diagram. Because each time chunk may have a different top-50-most-common stack trace, you may end
+up with a few hundred stack traces that are made explicit (e.g. not grouped into the "other"
+category).
+
+The number of traces collected at a given point in time is correlated to overall CPU load - an idle
+machine will generate fewer traces than a machine running at full load, as cores sent to sleep will
+not generate traces in Prodfiler.
+
+Clicking on one of the "30M", "1H", "24H", "7D", "30D" buttons will switch the web interface to
+display the current point in time minus 30 minutes, 1 hour, etc.
+
+Click-and-drag into the chart allows you to drill down on a particular timeframe. Single-click on
+a block in the diagram will pull up the last two stack frames associated with that trace, providing
+a bit more context -- a double-click will pop up a window with all stack frames of the trace.
+
+![traces single click screenshot](./pictures/traces-single-click.png)
+![traces double click screenshot](./pictures/traces-double-click.png)
+
 ## The Functions view
 
+Aside from inspecting the most common traces, you can also inspect the most common **functions**.
+Simply click on the "functions" button in the top navigation bar. The time interval you have
+already selected will stay the same.
+
+![top n functions screenshot](./pictures/top-n-functions.png)
+
+### Comparing the Top-N functions between two time intervals
+
+It can often be helpful to compare two timeframes or subsets of machines - identifying which
+functions got relatively more expensive or relatively less expensive. In the "differential Top-N"
+view, you can do just that: Specify two different time frames (or two different subsets of machines
+using the filter language), and see which functions got more expensive or less expensive.
+
+![top n functions comparison screenshot](./pictures/top-n-functions-comparison.png)
+
 ## The Flamegraph view
 
+Flamegraphs are a common way of visualizing collections of stack traces. By clicking on the
+"Flamegraph" button in the top navigation bar, the construction of a flamegraph is triggered on
+the backend, and the result will be loaded in the browser. **Please note that this is a somewhat
+heavy operation, and it can take 30s or longer to generate and load a flame graph**.
+
+The runtimes to which a particular stack frame belongs are marked by different colors: Brown-ish
+hues represent python, mint-ish hues represent native userspace code, green-ish hues are JVM, and
+blue-ish hues are from the Linux kernel.
```

Comment:
Could we also add the colors for Ruby and Perl stack frames? I think Ruby is red-ish.

Since this list could continue to grow, perhaps a table showing the stack frame and its associated color would be helpful for the reader.

Reply:
Yeah a color chart (or at least full reference) is a good idea.